python/tvm/relax/backend/dispatch_sort_scan.py

```diff
@@ -154,7 +154,48 @@ def visit_call_(self, call: relax.Call) -> relax.Expr:
         if call.op.name in ("relax.cumprod", "relax.cumsum"):
             tgt = self._get_target(call.struct_info)
             axis = int(call.attrs.axis) if call.attrs.axis is not None else call.attrs.axis
+            shape = call.struct_info.shape
             kwargs = {}
+            if (
+                (axis == -1 or axis == len(shape) - 1)
```

Comment:
For tensors of unknown shape, the `shape` field is none.  Instead of `len(call.struct_info.shape)`, can we use `call.struct_info.ndim`?  (Alternatively, since it looks like the implementation requires an explicit shape in order to apply a reshape, we could add `shape is not None` to this condition.)

Reply:
Thanks for the caching. Unfortunately, the original implementation does not support unknown shape. I added a check in the pass